### PR TITLE
Start making the network connection more robust

### PR DESCRIPTION
**Pull request by kasperl**
Retries, etc.

### CODE REVIEW

File: src/jaguar.toit [hunk 1]
```diff
@@ -125,3 +147,28 @@ identify id/uuid.Uuid name/string address/string -> none:
   while true:
     socket.send msg
     sleep --ms=200
+
+serve_incoming_requests socket/tcp.ServerSocket id/uuid.Uuid:
```
Comment (kasperl) [inline src/jaguar.toit:151]: Unchanged and lifted out of main.

File: src/jaguar.toit [hunk 2]
```diff
@@ -63,31 +67,56 @@ main args:
...
+  network/net.Interface? := null
+  error := null
+
+  try:
+    network = net.open
+    socket/tcp.ServerSocket := network.tcp_listen port
+    address := "http://$network.address:$socket.local_address.port"
+    logger.info "running Jaguar device '$name' (id: '$id') on '$address'"
+
+    // We run two tasks concurrently: One broadcasts the device identity
```
Comment (kasperl) [inline src/jaguar.toit:94]: This appears to be generally useful. Maybe we can factor this out and let other's benefit from this support for running multiple tasks that start/stop as one?

File: src/jaguar.toit [hunk 3]
```diff
@@ -63,31 +67,56 @@ main args:
...
+    // via UDP and one serves incoming HTTP requests. If one of the tasks
+    // fail, we take the other one down to clean up nicely.
+    done := monitor.Semaphore
+    server_task = task::
+      try:
+        error = catch: serve_incoming_requests socket id
+      finally:
+        server_task = null
+        if broadcast_task: broadcast_task.cancel
+        done.up
```
Comment (erikcorry) [inline src/jaguar.toit:104]: You might put this in yet another layer of try-finally so that it still happens if broadcast_task.cancel fails?

**Review by erikcorry — APPROVED** [review]

File: src/jaguar.toit (hunk 3 above)
Comment (kasperl) [reply src/jaguar.toit:104]: As discussed, I'd prefer to be able to count on cancel always working.

**Pull request merged**